Sources/StreamChat/Workers/TypingEventsSender.swift

```diff
@@ -39,6 +39,19 @@ class TypingEventsSender: Worker {
         guard currentUserLastTypingDate != nil else { return }
         self.stopTyping(in: typingInfo.channelId, parentMessageId: typingInfo.parentMessageId)
     }
+    
+    func currentUserTypingEventsState(completion: @escaping (Bool) -> Void) {
+        database.read { session in
+            session.currentUser?.isTypingIndicatorsEnabled ?? true
+        } completion: { result in
+            switch result {
+            case .success(let isEnabled):
+                completion(isEnabled)
+            case .failure:
+                completion(false)
+            }
+        }
+    }
```

Comment:
Like I said above, I think accessing the dataStore.currentUser() directly, should be enough. WDYT?